cpp/src/gandiva/function_registry_datetime.cc

```diff
@@ -155,6 +155,10 @@ std::vector<NativeFunction> GetDateTimeFunctionRegistry() {
           NativeFunction::kNeedsContext | NativeFunction::kNeedsFunctionHolder |
               NativeFunction::kCanReturnErrors),
 
+      NativeFunction("trunc", {}, DataTypeVector{date64(), utf8()}, utf8(),
+                     kResultNullIfNull, "date_trunc_timestamp_utf8",
```

Comment:
The parameter that you receive is a date, but you named the function using timestamp, change the name to date.